lib/statsig/logging.ex

```diff
@@ -0,0 +1,103 @@
+defmodule Statsig.Logging do
+  use GenServer
+  require Logger
+
+  defmodule State do
+    defstruct events: [], flush_timer: nil, flush_interval: 60_000
+
+    @type t :: %__MODULE__{
+            events: list(),
+            flush_timer: reference() | nil,
+            flush_interval: integer()
+          }
+
+    def new(opts \\ []) do
+      %__MODULE__{
+        events: [],
+        flush_timer: nil,
+        flush_interval: flush_interval()
+      }
+    end
+
+    def add_event(state, event) do
+      %{state | events: [event | state.events]}
+    end
+
+    def set_timer(state, timer) do
+      %{state | flush_timer: timer}
+    end
+
+    def clear_events(state) do
+      %{state | events: []}
+    end
+
+    def flush_events(%__MODULE__{events: []} = state), do: {state, []}
+
+    def flush_events(%__MODULE__{events: events} = state) do
+      {failed_events, successful_events} =
+        events
+        |> Enum.chunk_every(500)
+        |> Enum.reduce({[], []}, fn chunk, {failed, successful} ->
+          case api_client().push_logs(chunk) do
+            {:ok, _} ->
+              {failed, successful ++ chunk}
```

Comment:
`++` can be pretty slow, since you're just collecting the lists, inside the reduce, do
```elixir
{failed, [successful, chunk]}
```
or 
```elixir
{[failed, chunk], successful}
```
and do 
```elixir
{%{state | failed_events: List.flatten(failed_events)}, List.flatten(successful_events)}
```